src/foam/nanos/notification/NotificationSetting.js

```diff
@@ -62,7 +85,18 @@ foam.CLASS({
         notification.setUserId(user.getId());
         notification.setBroadcasted(false);
         notification.setGroupId(null);
-        notification.setEmailIsEnabled(false);
+
+        // We cannot permanently disable in-app notifications, so mark them read automatically
+        if ( ! getEnabled() ) {
+          notification.setRead(true);
+        }
+        else if ( user.getDisabledTopics() != null ) {
+          List disabledTopics = Arrays.asList(user.getDisabledTopics());
```

Comment:
This seems expensive. It will on average take 1.5*n operations plus generate a throw-away garbage List. Could we write a helper method to just find in the array, or better, have the javaPreSet for the property sort the array before storing then call Arrays.binarySearch() which will be log n? Same with above.

Reply:
i'll see what I can do.